beacon-chain/state/stategen/replay.go

```diff
@@ -318,34 +318,6 @@ func (s *State) archivedState(ctx context.Context, slot uint64) (*state.BeaconSt
 	return st, err
 }
 
-// This recomputes a state given the block root.
-func (s *State) recoverStateByRoot(ctx context.Context, root [32]byte) (*state.BeaconState, error) {
```

Comment:
@terencechain do we still want to keep this ?

Reply:
no longer needed after #6461